sysid-application/src/main/native/cpp/analysis/AnalysisManager.cpp

```diff
@@ -595,6 +596,12 @@ AnalysisManager::AnalysisManager(std::string_view path, Settings& settings,
 
   // Get the rotation -> output units factor from the JSON.
   m_unit = m_json.at("units").get<std::string>();
+  for (int i = 0; i < sizeof(kUnits); ++i) {
+    if (m_unit == kUnits[i]) {
+      m_abbrevUnit = GetAbbreviation(kUnits[i]);
+      break;
+    }
+  }
   m_factor = m_json.at("unitsPerRotation").get<double>();
```

Comment:
It might be worth reducing a little bit of code duplication by using the `ResetUnitsFromJSON` function here. The function would need a better name though

Reply:
Should PrepareData() be called in the constructor though?

Reply:
Ah it probably shouldn't so I that should be fine especially with the updated changes